src/handlers.js

```diff
@@ -0,0 +1,89 @@
+const DEFAULTS = {
+  create: {
+    action (component, req) {
+      return component.create(req.body, req.jsdataOpts)
+    },
+    statusCode: 201
+  },
+  createMany: {
+    action (component, req) {
+      return component.createMany(req.body, req.jsdataOpts)
+    },
+    statusCode: 201
+  },
+  destroy: {
+    action (component, req) {
+      return component.destroy(req.params.id, req.jsdataOpts)
+    },
+    statusCode: 204
+  },
+  destroyAll: {
+    action (component, req) {
+      return component.destroyAll(req.query, req.jsdataOpts)
+    },
+    statusCode: 204
+  },
+  find: {
+    action (component, req) {
+      return component.find(req.params.id, req.jsdataOpts)
+    },
+    statusCode: 200
+  },
+  findAll: {
+    action (component, req) {
+      return component.findAll(req.query, req.jsdataOpts)
+    },
+    statusCode: 200
+  },
+  update: {
+    action (component, req) {
+      return component.update(req.params.id, req.body, req.jsdataOpts)
+    },
+    statusCode: 200
+  },
+  updateAll: {
+    action (component, req) {
+      return component.updateAll(req.body, req.query, req.jsdataOpts)
+    },
+    statusCode: 200
+  },
+  updateMany: {
+    action (component, req) {
+      return component.updateMany(req.body, req.jsdataOpts)
+    },
+    statusCode: 200
+  }
+}
+
+export function makeRequestHandler (method, component, config = {}) {
+  const action = config[method].action || DEFAULTS[method].action
+
+  return (req, res, next) => {
+    action(component, req)
+      .then((result) => {
+        req.jsdataResult = result
+        next()
+      })
+      .catch(next)
+  }
+}
+
+export function makeResponseHandler (method, component, config = {}) {
+  const statusCode = config[method].statusCode || DEFAULTS[method].statusCode
```

Comment:
This will throw `Uncaught TypeError: Cannot read property 'statusCode' of undefined(…)` if `config[method]` is not defined.